quantum/pointing_device/pointing_device_scroll.h

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+#include "pointing_device.h"
+//#include "report.h"
+
+#ifndef POINTING_DEVICE_SCROLL_ENABLE
+#    error "POINTING_DEVICE_SCROLL_ENABLE not defined! settings may be incorrect!"
+#endif
+
+#ifndef SCROLL_TAP_DELAY
+#    define SCROLL_TAP_DELAY 0
+#endif
+
+#ifndef SCROLL_DRAG_DIVISOR
+#    define SCROLL_DRAG_DIVISOR 32
+#endif
+#ifndef SCROLL_CARET_DIVISOR
+#    define SCROLL_CARET_DIVISOR 32
+#endif
+#ifndef SCROLL_VOL_DIVISOR
+#    define SCROLL_VOL_DIVISOR 32
+#endif
+#ifndef SCROLL_HIST_DIVISOR
+#    define SCROLL_HIST_DIVISOR 64
+#endif
+
+/* local data structure */
+typedef struct {
+    uint8_t           mode;
+    uint8_t           mode_tg;
+    mouse_xy_report_t h;
+    mouse_xy_report_t v;
+} scroll_record_t;
+
+/* Default Scroll Modes */
+enum scroll_modes {
+    SM_NONE = 0x00, // Null scroll mode
+    SM_DRAG,
+    SM_CARET,
+    SM_VOLUME,
+    SM_HISTORY,
+    SM_SAFE_RANGE // to add more scroll modes leave at end
+};
+
+/* keycode macros */
+#define SM_MO(sm) sm > 0 ? (((sm)&0x0f) + SCROLL_MODE_MO_START - 1) : SCROLL_MODE_MO_START
+#define SM_TG(sm) sm > 0 ? (((sm)&0x0f) + SCROLL_MODE_TG_START - 1) : SCROLL_MODE_TG_START
+#define SM_TT(sm) sm > 0 ? (((sm)&0x0f) + SCROLL_MODE_TT_START - 1) : SCROLL_MODE_TT_START
```

Comment:
these should go into quantum_keycodes.h, so that keymaps won't fail to compile if the feature is disabled

Reply:
done and moved with some more descriptive comments similar to the layer macros

Reply:
The enum of scroll modes and aliases would also be needed as well then.